src/eth-signing/sign-off-chain-action.ts

```diff
@@ -130,11 +141,44 @@ export abstract class SignOffChainAction<M extends {}> extends Signer {
     expectedSigner: Address,
     message: M,
   ): boolean {
-    const hash = this.getHash(message);
-    const signer = ecRecoverTypedSignature(hash, typedSignature);
+    if (stripHexPrefix(typedSignature).length !== 66 * 2) {
+      throw new Error(`Unable to verify signature with invalid length: ${typedSignature}`);
+    }
+
+    const sigType = parseInt(typedSignature.slice(-2), 16);
```

Comment:
nit: do `stripHexPrefix(typedSignature)` once and feed the input into both the `if()` and the `parseInt()`

Reply:
I think that's not needed actually, you may have misread `typedSignature.slice(-2)` as `typedSignature.slice(2)`